binary_search_networks/test.py

```diff
@@ -0,0 +1,7 @@
+# Author: Isamu Isozaki
+# Date: 2020/11/10
+# Purpose: Take X, y and model as input and returns test accuracy
+
+def test(X, y, model):
+    _, accuracy = model.evaluate(X, y)
```

Comment:
Can you get accuracy, precision, recall, f1-score, and the confusion matrix as well?

Reply:
Yup. I'll do that in train.py too hopefully by the end of today.

Reply:
Here is how I did it for mine with `scikit-learn==0.20.4`. Feel free to modularize. 
```Python3
print("Start evalution of the model")

# Evaluting the model with different metrics
from sklearn.metrics import confusion_matrix
from sklearn.metrics import accuracy_score
from sklearn.metrics import auc
from sklearn.metrics import classification_report
from sklearn.metrics import roc_curve

# print(X_train.shape, X_test.shape, y_pred.shape, y_train.shape, y_test.shape)
acc_score = accuracy_score(y_test, y_pred)
fpr, tpr, thresholds = roc_curve(y_test, y_pred) 
area_under_curve = auc(fpr, tpr)
cr = classification_report(y_test, y_pred)
cm = confusion_matrix(y_test, y_pred)

print("Evaluation is over.")
print("#"*30)
print("Model Metrics")
print("Accuracy Score: ", acc_score)
print("Area Under Curve: ", area_under_curve)
print("Classification Report:\n", cr)
print("Confusion Matrix:\n", cm)
print("#"*30)
```

Reply:
Thank you! Referenced your code a bunch.